ee/clickhouse/queries/sessions/list.py

```diff
@@ -61,7 +61,9 @@ def fetch_page(self) -> Tuple[List[Session], Optional[Dict]]:
 
         self._add_person_properties(result)
 
-        return filter_sessions_by_recordings(self.team, result, self.filter), pagination
+        if self.filter.limit_by_recordings:
```

Comment:
Wouldn't doing this result in no session recordings being shown?

From the code:
> # :TRICKY: This mutates sessions list

Reply:
oof, yeah you are right. `filter_sessions_by_recordings` this does not. Basically `join_session_data` with `filter_sessions_by_recordings` optional